src/main.rs

```diff
@@ -42,59 +48,37 @@ enum Commands {
     },
 }
 
-fn create_fuzzy_finder(issues: Vec<String>) -> String {
-    // Create a list of items from the list of issues
-    let items: Vec<Item<String>> = issues
-        .iter()
-        .map(|issue| Item::new(issue.to_string(), issue.to_string()))
-        .collect();
-
-    // Prompt the user to select an issue from the list of issues
-    let items_count = items.len();
-
-    match fuzzy_finder::FuzzyFinder::find(items, items_count.try_into().unwrap()) {
-        Ok(Some(result)) => result,
-        Ok(None) => {
-            panic!("Invalid result");
-        }
-        Err(e) => {
-            std::io::stdout().flush().unwrap();
-            panic!("Failed to find result: {}", e);
-        }
-    }
-}
-
 fn main() {
-    // let issues = jira::jira_get_issues(args.jira_board_id);
-    // let issue_keys = jira::jira_get_issue_keys(&issues);
-    //
-    // let issue_key = fuzzy_finder::fuzzy_find(issue_keys, |item| item.as_str())
-    //     .expect("No issue selected")
-    //     .to_string();
-    //
-    // let issue = jira::jira_get_issue(&issue_key);
-    //
-    // let branch_name = format!(
-    //     "{}-{}",
-    //     issue_key,
-    //     jira::jira_get_issue_summary(&issue).replace(" ", "-")
-    // );
-    let issues = drivers::mock::get_issues("GBC");
-
     let opts: Opts = Opts::parse();
+    let mut config = Config::load().unwrap();
 
     match opts.command {
-        Commands::GlobalConfig { config } => {
-            println!("GlobalConfig");
-            println!("{:?}", config);
-        }
-        Commands::Init { config } => {
-            println!("Init");
-            println!("{:?}", config);
+        Commands::GlobalConfig { command } => match command {
+            GlobalConfigCommands::AddCredentials => {
+                let credentials = ui::get_credentials();
+                config.create_or_update_credentials(credentials).unwrap();
+            }
+            GlobalConfigCommands::Display => {
+                config.print();
+            }
+        },
+        Commands::Init {} => {
+            println!("Init was called");
         }
         Commands::New { path } => {
-            let item = create_fuzzy_finder(issues);
+            let credentials_url = &config
+                .get_credentials("https://jira.atlassian.com".to_string())
+                .unwrap()
+                .url;
+            let issues = drivers::mock::get_issues(&credentials_url);
+            let item = fuzzy_finder::create(issues);
             println!();
+
+            // let branch_name = format!(
+            //     "{}-{}",
+            //     issue_key,
+            //     jira::jira_get_issue_summary(&issue).replace(" ", "-")
+            // );
```

Comment:
This is here to remember me that I need to add some kind of branch formatting option with prefixes maybe give the option to add in the project file some available prefixes or types of branches with formatters. Something like:

```json
{ 
    "branch_types":  [
        { "name": "feature", "formatter": "feature/{issueID}-{issueDescription.dashed}" },
        { "name": "fix", "formatter": "fix/{issueID}-{issueDescription.dashed}" },
        { "name": "release", "formatter": "release/{issueID}-{issueDescription.dashed}" }
    ]
}
```

And then this gives you an option in the CLI

```shell
gcb new (feature | fix | release)
```